### PR TITLE
Allow layer elevation to be controlled via theme

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
Allows elevation to be added to layer via theme `layer.container.elevation` where any valid Box elevation values are accepted.

#### Where should the reviewer start?
src/js/components/Layer/LayerContainer.js

#### What testing has been done on this PR?
Added a jest test and tested local in storybook by creating a custom theme that merged the HPE theme. Applied:
```
const myTheme = deepMerge(hpe, {
   layer: {
      container: {
         elevation: 'large',
      },
   },
});
```
Screen Shot 2021-01-04 at 11 43 46 AM

#### How should this be manually tested?
With the above code snippet.

#### What are the relevant issues?
Related to https://github.com/grommet/grommet-theme-hpe/issues/149

#### Do the grommet docs need to be updated?
Yes, updated here.

#### Should this PR be mentioned in the release notes?
Yes. Can leverage theme to style Layer elevation using `theme.layer.container.elevation`

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/Layer/doc.js [hunk 1]
```diff
@@ -156,6 +156,12 @@ export const themeDoc = {
     type: 'string | (props) => {}',
     defaultValue: undefined,
   },
+  'layer.container.elevation': {
+    description: 'The amount of elevation. Any valid Box elevation value.',
```
Comment (ShimiSun) [inline src/js/components/Layer/doc.js:160]: I don't think that 'The amount of elevation' is clear.
Can we use something like 'Elevated height above the underlying container, indicated via a drop shadow. Any Box elevation value is valid.'

File: src/js/themes/base.d.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ import {
   BreakpointSize,
   ColorType,
   DeepReadonly,
+  ElevationType,
```
Comment (ShimiSun) [inline src/js/themes/base.d.ts:16]: Nice, thanks.

File: src/js/components/Layer/__tests__/Layer-test.js [hunk 3]
```diff
@@ -398,4 +398,21 @@ describe('Layer', () => {
     );
     expect(onClickOutside).toHaveBeenCalledTimes(1);
   });
+
+  test(`custom theme`, () => {
```
Comment (ShimiSun) [inline src/js/components/Layer/__tests__/Layer-test.js:402]:
```suggestion
  test('custom theme', () => {
```

**Review by ShimiSun — COMMENTED** [review]
Looks good to me, dropped a couple of tiny comments.

File: src/js/components/Layer/doc.js (hunk 1 above)
Comment (taysea) [reply src/js/components/Layer/doc.js:160]: Yes, I like that wording. I'll update now.

File: src/js/components/Layer/__tests__/Layer-test.js (hunk 3 above)
Comment (taysea) [reply src/js/components/Layer/__tests__/Layer-test.js:402]: Done.

**Review by ShimiSun — COMMENTED** [review]
Thank you 💟

**Pull request merged**